pdns/dnsdist-lua.cc

```diff
@@ -1987,6 +1988,19 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
   });
 
   luaCtx.writeFunction("setTCPInternalPipeBufferSize", [](uint64_t size) { g_tcpInternalPipeBufferSize = size; });
+  luaCtx.writeFunction("setTCPFastOpenKey", [](const std::string& keyString) {
+    setLuaSideEffect();
+    unsigned int key[4] = {};
+    auto ret = sscanf(keyString.c_str(), "%x-%x-%x-%x", &key[0], &key[1], &key[2], &key[3]);
```

Comment:
It might be better to use `uint32_t` instead of `unsigned int` since it seems to be what the kernel expect, and that might prevent some surprise on uncommon architectures.
```suggestion
    uint32 key[4] = {};
    auto ret = sscanf(keyString.c_str(), "%" SCNx32 "-%" SCNx32 "-%" SCNx32 "-%" SCNx32, &key[0], &key[1], &key[2], &key[3]);
```
And then we also need to change the type of `g_TCPFastOpenKey`.

Note that we likely need to add `#include <cinttypes>` at the top of that file for the `SCNx32` macro to be defined.

Reply:
Okay, it's all done.